caikit/core/toolkit/destroyable_process.py

```diff
@@ -175,9 +178,18 @@ def _update_result(self):
     @property
     def error(self) -> Optional[Exception]:
         self._update_result()
+
         if isinstance(self.__result, Exception):
             return self.__result
 
+        if self.exitcode and self.exitcode != os.EX_OK:
```

Comment:
I wonder if it makes more sense to put this in `self._update_result`? That way the value is stored and not recreated.

Reply:
@gabe-l-hart you mean storing MemoryError etc in `self.__result` ? hmm.  We can do that, but these would be static values only and the exit code etc is with the parent process, so in terms of processing, it would be equivalent.  Adding it in result, would result in 2nd time getting the error from `__result`, which might make a bit confusing, although practically, it would be same value, so may be this confusion would be unnoticeable